crates/typst/src/text/mod.rs

```diff
@@ -1300,3 +1304,26 @@ cast! {
         ret
     },
 }
+
+/// Function to parse the language argument.
+/// Provides a hint if a region is used in the language field.
+fn parse_lang(args: &mut Args) -> SourceResult<Option<Lang>> {
+    let Some(Spanned { v: iso, span }) = args.named::<Spanned<EcoString>>("lang")? else {
+        return Ok(None);
+    };
+
+    let result = Lang::from_str(&iso);
+    if result.is_err() && iso.contains('-') {
+        let pair: Vec<_> = iso.split('-').collect();
+        if pair.len() == 2
+            && Lang::from_str(pair[0]).is_ok()
+            && Region::from_str(pair[1]).is_ok()
+        {
```

Comment:
We could use `str::split_once()` here.

Reply:
Great, `split_once` is safer and more efficient here.